### PR TITLE
[VL] Support regr_sxx and regr_syy aggregate functions for Spark 3.4

### PR DESCRIPTION
**Pull request by liujiayi771**
## What changes were proposed in this pull request?

Support regr_sxx regr_syy aggregate functions.

## How was this patch tested?

Add new test cases.

### CODE REVIEW

**Comment by liujiayi771** [comment]
I will try to fix all these regr_xx aggregate functions for spark 3.5 in another PR.

**Comment by liujiayi771** [comment]
cc @PHILO-HE, thanks.

**Comment by liujiayi771** [comment]
cc @rui-mo @Yohahaha, thanks.

**Review by rui-mo — COMMENTED** [review]
Thanks, looks good.

> I will try to fix all these regr_xx aggregate functions for spark 3.5 in another PR.

Is there any unresolved issue for Spark 3.5?

**Comment by liujiayi771** [comment]
@rui-mo 
> Is there any unresolved issue for Spark 3.5?

Currently, it's unclear to me. I will review these functions together in Spark 3.5 and will find out which functions can be enabled. In theory, there should be no issues with any of them.

File: backends-velox/src/test/scala/org/apache/gluten/execution/VeloxAggregateFunctionsSuite.scala [hunk 1]
```diff
@@ -432,14 +432,46 @@ abstract class VeloxAggregateFunctionsSuite extends VeloxWholeStageTransformerSu
   }
 
   // Disable for Sparke3.5.
-  testWithSpecifiedSparkVersion("regr_sxy", Some("3.4"), Some("3.4")) {
+  testWithSpecifiedSparkVersion("regr_sxy regr_sxx regr_syy", Some("3.4"), Some("3.4")) {
```
Comment (rui-mo) [inline backends-velox/src/test/scala/org/apache/gluten/execution/VeloxAggregateFunctionsSuite.scala:435]: Have we tried this test on 3.5, and what is the behavior?

Comment (rui-mo) [reply backends-velox/src/test/scala/org/apache/gluten/execution/VeloxAggregateFunctionsSuite.scala:435]: Got it. Thanks.

**Review by rui-mo — APPROVED** [review]

**Pull request merged**